### PR TITLE
fix: Search For Words With Less Than Three Chars

### LINKED ISSUE #2799: 🐛 [Bug] Search input with a space ignores minimal character length
> [sushidave](https://github.com/sushidave) **Authored by [sushidave](https://github.com/sushidave)**
_<time datetime="2020-01-16T11:14:14Z" title="Thursday, January 16th 2020, 12:14:14 pm +01:00">Jan 16, 2020</time>_
_Closed <time datetime="2020-03-18T20:40:10Z" title="Wednesday, March 18th 2020, 9:40:10 pm +01:00">Mar 18, 2020</time>_
---

## :bug: Bugreport

If the search input contains at least two words separated by a space the search results contain matches that ignore the minimal character length.

Example 1:
hc-bug-seach-with-space-example-1

Example 2:
hc-bug-seach-with-space-example-2

### Steps to reproduce the behavior
1. In the search field, type in at least two words with a space in between, see the examples.
2. The search results show also records that only match the single character after the space.

### Expected behavior

Search results should be more accurate: "martina s" should return results that contain _martina AND s_ instead of _martina OR s_.

### Version & Environment
 Type: [any]      
 - OS: [any]      
 - Browser: [any] 
 - Version [any]  
 - Device: [any]

### PR DESCRIPTION
**Pull request by Tirokk**
> [Mogge](https://github.com/Mogge) **Authored by [Mogge](https://github.com/Mogge)**
_<time datetime="2020-02-06T21:45:33Z" title="Thursday, February 6th 2020, 10:45:33 pm +01:00">Feb 6, 2020</time>_
_Closed <time datetime="2020-03-10T11:20:02Z" title="Tuesday, March 10th 2020, 12:20:02 pm +01:00">Mar 10, 2020</time>_
---

## 🍰 Pullrequest

Hyphens are replaced by one space, which also fixes the hyphen search, this time in the frontend.
If the search string contains spaces, only words with at least three chars are passed to the backend.

### Issues
- fixes #2799

### CODE REVIEW

**Comment by Tirokk** [comment]
> [cypress[bot]](https://github.com/apps/cypress) **Authored by [cypress[bot]](https://github.com/apps/cypress)**
_<time datetime="2020-02-06T22:14:11Z" title="Thursday, February 6th 2020, 11:14:11 pm +01:00">Feb 6, 2020</time>_

---



<p align="center">
  <br />
  
</p>

---

### Test summary
<a href="https://dashboard.cypress.io/projects/qa7fe2/runs/4933/specs"></a> <b title="Passed">59</b>  &bull; <a href="https://dashboard.cypress.io/projects/qa7fe2/runs/4933/failures"></a> <b title="Failed">0</b>  &bull;  <b title="Pending">0</b> &bull;  <b title="Skipped">0</b>

---

### Run details
| | |
| --- | --- |
| **Project** | Human-Connection |
| **Status** | Passed |
| **Commit** | 5330262a5a |
| **Started** | Feb 6, 2020 10:04 PM |
| **Ended** | Feb 6, 2020 10:18 PM |
| **Duration** | 14:43 <a href="https://on.cypress.io/parallelization" title="Enable parallelization to reduce run duration">💡</a> |
| **OS** | Linux Ubuntu Linux - 16.04 |
| **Browser** | Electron 78  |

[View run in Cypress Dashboard ➡️](https://dashboard.cypress.io/projects/qa7fe2/runs/4933/specs)



---

This comment has been generated by **cypress-bot** as a result of this project's GitHub integration settings. You can manage this integration in [this project's settings in the Cypress Dashboard](https://dashboard.cypress.io/projects/qa7fe2/settings)

File: webapp/components/generic/SearchableInput/SearchableInput.vue [hunk 1]
```diff
@@ -89,9 +89,18 @@ export default {
...
+      if (isEmpty(this.value)) {
+        return
+      }
+      if (this.value.includes(' ')) {
+        this.value = this.value
+          .split(' ')
+          .filter(word => word.length > 2)
+          .join(' ')
+      }
+      if (this.value.length < 3) {
```
Comment (Tirokk) [inline webapp/components/generic/SearchableInput/SearchableInput.vue:103]: > [roschaefer](https://github.com/roschaefer) **Authored by [roschaefer](https://github.com/roschaefer)**
_<time datetime="2020-02-07T11:04:58Z" title="Friday, February 7th 2020, 12:04:58 pm +01:00">Feb 7, 2020</time>_

---
> **Outdated (history rewrite)** - original diff
---
```diff
@@ -89,9 +89,18 @@ export default {
     },
     handleInput(event) {
       clearTimeout(this.searchProcess)
-      this.value = event.target ? event.target.value.replace(/\s+/g, ' ').trim() : ''
+      this.value = event.target ? event.target.value.replace(/[-\s]+/g, ' ').trim() : ''
       this.unprocessedSearchInput = this.value
-      if (isEmpty(this.value) || this.value.replace(/\s+/g, '').length < 3) {
+      if (isEmpty(this.value)) {
+        return
+      }
+      if (this.value.includes(' ')) {
+        this.value = this.value
+          .split(' ')
+          .filter(word => word.length > 2)
+          .join(' ')
+      }
+      if (this.value.length < 3) {
```


This is a quite hacky solution. I'm going to approve this but I feel bad about it. Do you know "stopwords"? It's a feature of search indices and I think that would have been the proper solution.

@Mogge if you feel like it, you could add an integration test for it to document this behaviour at least.

**Comment by Tirokk** [comment]
> [roschaefer](https://github.com/roschaefer) **Authored by [roschaefer](https://github.com/roschaefer)**
_<time datetime="2020-02-07T11:06:13Z" title="Friday, February 7th 2020, 12:06:13 pm +01:00">Feb 7, 2020</time>_

---


Not gonna merge this to give @Mogge time to add tests if he feels like it.

**Comment by Tirokk** [comment]
> [Lulalaby](https://github.com/Lulalaby) **Authored by [Lulalaby](https://github.com/Lulalaby)**
_<time datetime="2020-02-22T05:26:18Z" title="Saturday, February 22nd 2020, 6:26:18 am +01:00">Feb 22, 2020</time>_

---


Seems like he don't want it.
Merging?

**Comment by Tirokk** [comment]
> [Mogge](https://github.com/Mogge) **Authored by [Mogge](https://github.com/Mogge)**
_<time datetime="2020-03-10T11:19:55Z" title="Tuesday, March 10th 2020, 12:19:55 pm +01:00">Mar 10, 2020</time>_

---


Better solution available

**Review by Mogge — APPROVED** [review]
meh

**Pull request closed without merging**